src/components/course-calendar/plan.cjsx

```diff
@@ -120,15 +130,22 @@ CoursePlan = React.createClass
     @setState(publishState)
 
   componentWillMount: ->
-    @subscribeToPublishing(@props.item.plan)
-    @syncRoute()
+    {plan} = @props.item
+    publishState = PlanHelper.subscribeToPublishing(plan, @checkPublishingStatus)
+    isViewingStats = @_doesPlanMatchesRoute()
+
+    state = _.extend({isViewingStats}, publishState)
+    @setState(state)
+
     location = @context.router.getLocation()
     location.addChangeListener(@syncRoute)
 
   componentWillReceiveProps: (nextProps) ->
     if @props.item.plan.id isnt nextProps.item.plan.id
-      @subscribeToPublishing(nextProps.item.plan)
+      publishState = PlanHelper.subscribeToPublishing(nextProps.item.plan, @checkPublishingStatus)
       @stopCheckingPlan(@props.item.plan)
+      state = _.extend(@getStateByProps(nextProps), publishState)
```

Comment:
it subscribes to the new plan, and then unsubscribes to the old plan, and only if it passes through [`@props.item.plan.id isnt nextProps.item.plan.id`](https://github.com/openstax/tutor-js/pull/1136/files#diff-60a357506dabb2d734615a51de235068L129)